src/librustc_platform_intrinsics/lib.rs

```diff
@@ -37,60 +36,41 @@ pub enum IntrinsicDef {
     Named(&'static str),
 }
 
+
+static VOID: Type = Type::Void;
+
 static I8: Type = Type::Integer(true, 8, 8);
 static I16: Type = Type::Integer(true, 16, 16);
 static I32: Type = Type::Integer(true, 32, 32);
 static I64: Type = Type::Integer(true, 64, 64);
-static U8: Type = Type::Integer(false, 8, 8);
-static U16: Type = Type::Integer(false, 16, 16);
-static U32: Type = Type::Integer(false, 32, 32);
-static U64: Type = Type::Integer(false, 64, 64);
```

Comment:
>That doesn't mean the Rust interface to those intrinsics should ignore Rust's richer type systems. Requiring wrapper crates to insert lots of ugly `transmute` would be a pointless complication IMHO,

Nothing is ignoring Rust's type system. Rust ignores Rust's type system at the `extern platform-instrict` layer. This is where you can inject `u64x2( u64,u64);` unsigned types perfectly safely then any interaction will be type safe to the `u64x2` type. 

If is is that big of a deal I'll re-add the unsigned types.